### PR TITLE
feat: Add converter support for logical_and

### PR DESCRIPTION
**Pull request by mfeliz-cruise**
# Description

Adds converter support for aten::logical_and
https://pytorch.org/docs/stable/generated/torch.logical_and.html

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Review by narendasan — COMMENTED** [review]
Do the tests test both cases where we expect true and false here?

**Comment by mfeliz-cruise** [comment]
> Do the tests test both cases where we expect true and false here?

Yes, both paths are exercised.
